### PR TITLE
Course detail - Articles tab migration to  mvvm

### LINKED ISSUE #319: [sub-issue] Migrate Course details page - Article tab to MVVM
**Migrate Course details page - Article tab to MVVM**

Currently, this page is developed using MVP architecture this should be migrated to MVVM

**Would you like to work on the issue?**
yes

### PR DESCRIPTION
**Pull request by gconnect**
## What this PR does
Fixes #319

**Describe the purpose of this pull request and note the issue it addresses**
- This PR migrates the course detail Article codebase from MVP to MVVM

### CODE REVIEW

**Comment by gconnect** [comment]
@ujjwalagrawal17 please review

File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/coursedetail/articlesedited/viewmodel/ArticlesEditedViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        _progressbar.postValue(false)
+
+        viewModelScope.launch {
+            try {
+                _articleList.postValue(articlesEditedRepository.requestArticlesEdited(url))
+            } catch (io: IOException) {
+                _showMsg.postValue(ShowMessage("Unable to connect to the server."))
+            }
+        }
+    }
```
Comment (ujjwalagrawal17) [inline app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/coursedetail/articlesedited/viewmodel/ArticlesEditedViewModel.kt]: I am unable to see the progressbar during loading. 
I think we need to modify it as below. 


```
        _progressbar.postValue(true)
 viewModelScope.launch {
            try {
                _articleList.postValue(articlesEditedRepository.requestArticlesEdited(url))
                _progressbar.postValue(false)
            } catch (io: IOException) {
                _showMsg.postValue(ShowMessage("Unable to connect to the server."))
                _progressbar.postValue(false)
            }
        }
```

**Comment by ujjwalagrawal17** [comment]
Looks good to me, just 1 minor change.

File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/coursedetail/articlesedited/viewmodel/ArticlesEditedViewModel.kt (hunk 1 above)
Comment (gconnect) [reply app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/coursedetail/articlesedited/viewmodel/ArticlesEditedViewModel.kt]: okay

**Comment by gconnect** [comment]
@ujjwalagrawal17 changes effected. Please review.

**Comment by ujjwalagrawal17** [comment]
Okay.

**Pull request merged**